apps/src/applab/designElements/photochooser.jsx

```diff
@@ -0,0 +1,226 @@
+          initialValue={parseInt(element.style.top, 10)}
+          handleChange={this.props.handleChange.bind(this, 'top')}
+        />
+        <ColorPickerPropertyRow
+          desc={'background color'}
+          initialValue={elementUtils.rgb2hex(element.style.backgroundColor)}
+          handleChange={this.props.handleChange.bind(this, 'backgroundColor')}
+        />
+        {iconColorPicker}
+        <BorderProperties
+          element={element}
+          handleBorderWidthChange={this.props.handleChange.bind(
+            this,
+            'borderWidth'
+          )}
+          handleBorderColorChange={this.props.handleChange.bind(
+            this,
+            'borderColor'
+          )}
+          handleBorderRadiusChange={this.props.handleChange.bind(
+            this,
+            'borderRadius'
+          )}
+        />
+        <BooleanPropertyRow
+          desc={'hidden'}
+          initialValue={$(element).hasClass('design-mode-hidden')}
+          handleChange={this.props.handleChange.bind(this, 'hidden')}
+        />
+        <ZOrderRow
+          element={this.props.element}
+          onDepthChange={this.props.onDepthChange}
+        />
+      </div>
+    );
+  }
+}
+
+class PhotoChooserEvents extends React.Component {
+  static propTypes = {
+    element: PropTypes.instanceOf(HTMLElement).isRequired,
+    handleChange: PropTypes.func.isRequired,
+    onInsertEvent: PropTypes.func.isRequired
+  };
+
+  getPhotoselectEventCode() {
+    const id = elementUtils.getId(this.props.element);
+    const code =
+      'onEvent("' +
+      id +
+      '", "change", function() {\n' +
+      '  console.log("' +
+      id +
+      ' photo selected!");\n' +
+      '  console.log(getImageURL("' +
+      id +
+      '"));\n' +
+      '});\n';
+    return code;
+  }
+
+  insertPhotoselect = () =>
+    this.props.onInsertEvent(this.getPhotoselectEventCode());
+
+  render() {
+    const element = this.props.element;
+    const clickName = 'Change';
+    const clickDesc = 'Triggered when a photo is selected.';
+
+    return (
+      <div id="eventRowContainer">
+        <PropertyRow
+          desc={'id'}
+          initialValue={elementUtils.getId(element)}
+          handleChange={this.props.handleChange.bind(this, 'id')}
+          isIdRow={true}
+        />
+        <EventHeaderRow />
+        <EventRow
+          name={clickName}
+          desc={clickDesc}
+          handleInsert={this.insertPhotoselect}
+        />
+      </div>
+    );
+  }
+}
+
+export default {
+  PropertyTab: PhotoChooserProperties,
+  EventTab: PhotoChooserEvents,
+  themeValues: themeValues.photoSelect,
+
+  create: function() {
+    const element = document.createElement('label');
+    element.setAttribute('class', 'img-upload');
+    element.setAttribute('data-canonical-image-url', 'icon://fa-camera');
+    element.style.margin = '0px';
+    element.style.lineHeight = '1';
+    element.style.overflow = 'hidden';
+    element.style.wordWrap = 'break-word';
```

Comment:
I don't think you need to worry about wordWrap here since the user can't set the element's text